tests/link/pytorch/test_nlinalg.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+import pytest
+
+from pytensor.compile.function import function
+from pytensor.configdefaults import config
+from pytensor.graph.fg import FunctionGraph
+from pytensor.tensor import nlinalg as pt_nla
+from pytensor.tensor.type import matrix
+from tests.link.pytorch.test_basic import compare_pytorch_and_py
+
+
+@pytest.fixture
+def matrix_test():
+    rng = np.random.default_rng(213234)
+
+    M = rng.normal(size=(3, 3))
+    test_value = M.dot(M.T).astype(config.floatX)
+
+    x = matrix("x")
+    return (x, test_value)
+
+
+@pytest.mark.parametrize(
+    "func",
+    (
+        pt_nla.eig,
+        pt_nla.eigh,
+        pt_nla.slogdet,
+        pytest.param(
+            pt_nla.inv, marks=pytest.mark.xfail(reason="Blockwise not implemented")
+        ),
+        pytest.param(
+            pt_nla.det, marks=pytest.mark.xfail(reason="Blockwise not implemented")
+        ),
```

Comment:
Why is Blockwise involved? Shouldn't be when testing the core case (matrix)

Reply:
Can you `dprint` the reference pytensor function (python mode) that is also compiled in `comparepytorch_and_py`?

Reply:
I was using in the tests instances like [pytensor.tensor.nlinalg.inv](https://github.com/pymc-devs/pytensor/blob/a6b9585e5ca2755be8c0194f03dfab0beff2b3f4/pytensor/tensor/nlinalg.py#L161-L162) which is an instance of `Blockwise`. I changed to the default constructor (i.e., `MatrixInverse` in this particular case). Does it make sense?

Reply:
Shouldn't matter, if there are no batch dimensions the Blockwise should be automatically removed before we try to convert to pytorch

Reply:
That  I am afraid is not happening.

Reply:
Can you pasteurized `fn.dprint(print_type=True)` where fn is the python backend function that is being compared to the pytorch one?

Reply:
When using `Matrixinverse` 

```
MatrixInverse [id A] <Matrix(float64, shape=(?, ?))>
 └─ x [id B] <Matrix(float64, shape=(?, ?))>
```

when using `inv`:
```
Blockwise{MatrixInverse, (m,m)->(m,m)} [id A] <Matrix(float64, shape=(?, ?))>
 └─ x [id B] <Matrix(float64, shape=(?, ?))>
```

Reply:
Is that the dprint of the compiled function? It's missing the order which shows when you dprint a function (and not a variable).

Reply:
Ok so the python reference function is doing fast_compile which won't have the rewrite, but torch should be doing fast_run which should

Reply:
After passing it through `function` I get:

```
# Matrix Inverse
MatrixInverse [id A] <Matrix(float64, shape=(?, ?))> 0
 └─ x [id B] <Matrix(float64, shape=(?, ?))>

# pinv
MatrixInverse [id A] <Matrix(float64, shape=(?, ?))> 0
 └─ x [id B] <Matrix(float64, shape=(?, ?))>

```
Is that what you meant?

Reply:
Yup, so Blockwise is removed and shouldn't be a reason for the test to fail

Reply:
> Ok so the python reference function is doing fast_compile which won't have the rewrite, but torch should be doing fast_run which should

here you've lost me a bit :wink:

Reply:
> Yup, so Blockwise is removed and shouldn't be a reason for the test to fail

You are absolutely right. I checked again and it is not failing. But I saw (or I believed I did) failing it at some point, that's why I added those waivers. Anyhow. Should I leave it like this?

Reply:
I will revisit the second linalg PR.

Reply:
You should remove the xfail

Reply:
I did.

Reply:
Sorry you don't have xfail because you're testing the Op directly...

That should be fine, Are the others the direct Op (not the blockwise) as well?

Reply:
Sorry, I've forgot to push that one.

Reply:
Something else that needs to be addressed on this one?